### ISSUE TITLE
Countdown reproduction on A100 completes, but accuracy is far below paper results

### ISSUE THREAD

**Issue by yuvrajpant56** [divelab/E2H-Reasoning#4]
Hi, thank you for open-sourcing this repository.

I am trying to reproduce the Countdown results reported in the paper, but I am getting much lower accuracy than reported, even after completing training and evaluation successfully on an A100 node.

I wanted to ask whether I am missing an important dependency, launch setting, or implementation detail for Countdown reproduction.

Setup

I used:
```bash
1 node with 2× A100 80GB

GPU 0: vLLM server

GPU 1: training/evaluation

conda environment from the repo

trained and tested on the countdown task
```
Model / task
```bash
model: Qwen/Qwen2.5-1.5B-Instruct

task: countdown

curriculum: gaussian

mu_exp = 0.5

sigma = 0.5
```
Important note
```bash
On my system, I could not use flash_attention_2 directly because of environment/package issues, so I used:

model.args.attn_implementation=sdpa

I kept bf16 on A100.
```

Commands used
vLLM server on GPU 0
```bash
CUDA_VISIBLE_DEVICES=0 \
trl vllm-serve \
  --model Qwen/Qwen2.5-1.5B-Instruct \
  --dtype bfloat16 \
  --max_model_len 4096 \
  --trust_remote_code true \
  --log_level warning
```

training on GPU 1
```bash
CUDA_VISIBLE_DEVICES=1 \
accelerate launch \
  --mixed_precision bf16 \
  --num_processes 1 \
  --num_machines 1 \
  --dynamo_backend no \
  main.py \
  mode=train \
  model=qwen1.5b \
  task=countdown \
  model.args.attn_implementation=sdpa \
  algorithm.args.output_dir=outputs/Qwen-1.5B_Countdown_gaussian_a100_clean2 \
  algorithm.args.logging_dir=outputs/Qwen-1.5B_Countdown_gaussian_a100_clean2 \
  algorithm.e2h_args.curriculum_schedule=gaussian \
  algorithm.e2h_args.scheduler_args.mu_exp=0.5 \
  algorithm.e2h_args.scheduler_args.sigma=0.5
```
evaluation on GPU 1:
```bash
CUDA_VISIBLE_DEVICES=1 \
accelerate launch \
  --mixed_precision bf16 \
  --num_processes 1 \
  --num_machines 1 \
  --dynamo_backend no \
  main.py \
  mode=test \
  model=qwen1.5b \
  task=countdown \
  model.args.attn_implementation=sdpa \
  algorithm.args.output_dir=outputs/Qwen-1.5B_Countdown_gaussian_a100_clean2 \
  algorithm.args.logging_dir=outputs/Qwen-1.5B_Countdown_gaussian_a100_clean2 \
  algorithm.e2h_args.curriculum_schedule=gaussian \
  algorithm.e2h_args.scheduler_args.mu_exp=0.5 \
  algorithm.e2h_args.scheduler_args.sigma=0.5
```
Result I obtained
```bash
My test_results.json is:

{
  "overall": {
    "accuracy": 0.013671875,
    "support": 1024
  },
  "level_0": {
    "accuracy": 0.03535353535353535,
    "support": 198
  },
  "level_1": {
    "accuracy": 0.027522935779816515,
    "support": 218
  },
  "level_2": {
    "accuracy": 0.0048543689320388345,
    "support": 206
  },
  "level_3": {
    "accuracy": 0.0,
    "support": 199
  },
  "level_4": {
    "accuracy": 0.0,
    "support": 203
  }
}
```

What I expected

From the paper, I expected Countdown performance for the Gaussian E2H setup to be much higher than the above, especially on lower difficulty levels.

Additional observations

training completed successfully to 1600/1600

adapter was saved successfully

evaluation also completed successfully

in test_outputs.jsonl, many outputs seem to get only the format reward and fail correctness

during training I also saw the warning:

None of the inputs have requires_grad=True. Gradients will be None

however, step counts were progressing and training completed

Question

Could you please clarify whether one of the following is required for correct reproduction?

1. a specific flash_attn setup rather than sdpa

2. a different launch command than the one above

3. a missing dependency or environment setting

If helpful, I can also share:

---main.log

---a few entries from test_outputs.jsonl

---environment/package versions

Thank you.

Since training and evaluation both complete successfully but the resulting accuracy is far below the paper, I am wondering whether there is a reproduction issue, a missing undocumented dependency, or a mismatch between the released code and the reported setup.

**Comment by shubhamprshr27** [comment]
We have seen these problems with latest versions of TRL on our main branch. Since our code was developed before GRPO was standardized in TRL.

However, our (diriter) code in branch (messy) was working until ICLR. It is a bit un-organized but works and has been tested thoroughly. I would recommend to use that for reproduction, the script for doing so can be found in - 

https://github.com/divelab/E2H-Reasoning/blob/messy/run_hprc_tamu.slurm

It has all the details necessary. We will be releasing our models on HF Pages soon, but the code cleanup will take sometime to complete, but in the meanwhile feel free to use our other branch.

**Comment by shubhamprshr27** [comment]
Please let us know if you face any issues with our other branch, I will close this issue for now.
